en/ros/mavros_custom_messages.md

```diff
@@ -0,0 +1,348 @@
+# Sending Custom Message from MAVROS to PX4
+
+> **Warning** This article is not compatible with current development environment/stable versions. 
+  It has been tested against:
+  - **Ubuntu:** 14.04
+  - **ROS:** Kinetic
+  - **PX4 Firmware:** 1.6.2 dev, px4fmu-v2_default
+
+<!-- Content reproduced with permission from @JoonmoAhn in https://github.com/JoonmoAhn/Sending-Custom-Message-from-MAVROS-to-PX4/issues/1 -->
+
+## MAVROS Installation
+
+Follow *Source Installation* instructions from [mavlink/mavros](https://github.com/mavlink/mavros/blob/master/mavros/README.md) to install "ROS Kinetic".
+
+## MAVROS Changes
+
+1. Create **keyboard_command.cpp** as a mavros plugin (in **workspace/src/mavros/mavros_extras/src/plugins**) and copy in the code below.
+   The code subscribes a 'char' message from ROS topic `/mavros/keyboard_command/keyboard_sub` and sends it as a MAVLink message.
+   ```c
+   #include <mavros/mavros_plugin.h>
+   #include <pluginlib/class_list_macros.h>
+   #include <iostream>
+   #include <std_msgs/Char.h>
+
+   namespace mavplugin {
+
+   class KeyboardCommandPlugin : public MavRosPlugin {
+   public:
+       KeyboardCommandPlugin() :
+           nh("~keyboard_command"),
+           uas(nullptr)
+       { };
+
+       void initialize(UAS &uas_)
+       {
+           uas = &uas_;
+           keyboard_sub = nh.subscribe("keyboard_sub", 10, &KeyboardCommandPlugin::keyboard_cb, this);
+       };
+
+       const message_map get_rx_handlers() {
+           return {/* RX disabled */ };
```

Comment:
To be a complete example on how to create a mavros plugin, there could be a general exaple on how to sends a message from the FCU to ROS and how to create a custom mavros message